base/docs/helpdb/Base.jl

```diff
@@ -2317,3 +2317,57 @@ for bit in (8, 16, 32, 64, 128)
         $(Symbol("UInt", bit))
     end
 end
+
+"""
+    Vector{T}(len)
+
+Construct an uninitialized [`Vector{T}`](@ref) of length `len`.
+
+# Examples
+```julia-repl
+julia> Vector{Float64}(3)
+3-element Array{Float64,1}:
+ 6.90966e-310
+ 6.90966e-310
+ 6.90966e-310
+```
+"""
+Vector{T}(len)
+
+"""
+    Matrix{T}(m, n)
+
+Construct an uninitialized [`Matrix{T}`](@ref) of size `m`×`n`.
+
+# Examples
+```julia-repl
+julia> Matrix{Float64}(2, 3)
+2×3 Array{Float64,2}:
+ 6.93517e-310  6.93517e-310  6.93517e-310
+ 6.93517e-310  6.93517e-310  1.29396e-320
+```
+"""
+Matrix{T}(m, n)
+
+"""
+    Array{T}(dims)
+    Array{T,N}(dims)
+
+Construct an uninitialized [`Array{T,N}`](@ref).
+`N` is determined from the length or number of `dims`. `dims` may
+be a tuple or a series of integer arguments corresponding to the lengths in each dimension.
+If the rank `N` is supplied explicitly as in `Array{T,N}(dims)`, then it must
+match the length or number of `dims`.
+
+# Examples
+```jldoctest
+julia> A = Array{Float64,2}(2, 2);
```

Comment:
Updated. Removed `ndims` and `eltype` since I added them in #22052 mainly to show the resulting `N` and `T` when the output of the `Array` call was suppressed. Also ref https://github.com/JuliaLang/julia/pull/22052#issuecomment-303855834 so added one example where `N` is determined from the input.